### PR TITLE
OSD-13300 Migrate to OperatorSDK v1.21

### PR DESCRIPTION
**Pull request by mrbarge**
## What

This PR migrates `managed-velero-operator` to be compatible with OperatorSDK 1.x standards.

## Notes

There are some caveats with the PR:

- `vmware-tanzu/velero` is presently incompatible with k8s dependencies of 0.24 [0]. The k8s dependencies have been left matching the existing operator's versions. This similarly impacts `controller-runtime` and the version of `operator-lib` that can be used. This unfortunately means this operator cannot currently align version-wise with other managed OpenShift operators.

- OperatorSDK 1.x reworks the way that metrics on the CR are generated, deprecating a lot of the existing code. This now uses the `InstrumentedEnqueueRequestForObject` function to do the same job. This means that the metric changes from:

```
veleroinstall_info{namespace="openshift-velero",veleroinstall="cluster"} 1
```

to

```
resource_created_at_seconds{group="managed.openshift.io",kind="VeleroInstall",name="cluster",namespace="openshift-velero",version="v1alpha2"} 1.664360242e+09
```

Hosting the CR metrics from a separate port has been removed as a result. (all metrics are served from `8383`).

Refs: [SDE-1972](https://issues.redhat.com//browse/SDE-1972) (was not captured in original scope, but can't be neglected)

[0] https://github.com/vmware-tanzu/velero/issues/5009
[1] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#export-metrics

### CODE REVIEW

**Review by yithian — APPROVED** [review]
/lgtm

seems sane to me

File: .gitattributes [hunk 1]
```diff
@@ -1,5 +1,3 @@
-boilerplate/** linguist-generated=true
```
Comment (cblecker) [inline .gitattributes:1]: Why is this being deleted?

Comment (mrbarge) [reply .gitattributes]: I think this was a casualty as part of the SDKv1 migration process which involves re-onboarding the operator into a fresh boilerplate. But, should not have been dropped intentionally - I've pushed a commit which restores this configuration.

**Comment by cblecker** [comment]
/lgtm
/approve

Let's let this bake in stage! Thanks @mrbarge!

**Pull request merged**